---
title: "2017년 상반기 블로깅 회고를 위한 텍스트마이닝 (2) konlpy test"
layout: post
date: 2017-09-19 15:58
image: /assets/images/post/002/169_00title.png
headerImage: true
category: Blog
tag:
- 파이썬
- 텍스트마이닝
blog: true
author: Hyeyeon
description: 2017년 상반기 블로깅 회고를 위한 텍스트마이닝 (2) konlpy test
---

### Summary:

[2017년 상반기 블로깅 회고를 위한 텍스트마이닝 (1) 실패로그](https://imyeonn.github.io/blog/blog/169/)에서 이어집니다.

---

## 성공!!!!!

![pic1](/assets/images/post/002/173_01.png)

잠도 못 자고 며칠 동안 삽질한 결과 속썩이던 konlpy 임포트 문제가 드디어 해결됐다.

<br>

![pic2](/assets/images/post/002/173_02.png)

재부팅해도 `import konlpy`가 된다. 감동.

<br>

---

## 야 이 배신자야

![pic3](/assets/images/post/002/173_03.png)
<figcaption class="caption">ModuleNotFoundError: No module named 'konlpy.tag'; 'konlpy' is not a package</figcaption>

<br>

코엔엘파이 임포트가 잘 되는 걸 확인하고 며칠동안 파이썬을 묵혀뒀다. 이게 문제였던걸까. 오랜만에 터미널을 켜서 `import konlpy` 하니까 아래와 같은 오류 메세지가 떴다.

```
>>> import konlpy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/hyeyeonyang/konlpy.py", line 3, in <module>
    from konlpy.tag import Twitter
```

<figcaption class="caption">신이시여..</figcaption>

<br>

한참 당황하다가 제일 만만한 `import re`를 해보니 이건 또 된다. 어 혹시? 싶어서 konlpy를 다시 import해봤는데 이번엔 또 되네... 뭐지? 첫 번째와 두 번째 시도 사이 설정을 바꾼다거나 재부팅을 한다거나 터미널을 껐다 켰다거나, 그런 거 안했는데ㅠㅠ

일단 되긴 됐으니까 [저번에 포기했던 간단한 코드](https://imyeonn.github.io/blog/blog/169/)를 테스트해봤다.

{% highlight python %}
from konlpy.tag import Twitter

sentence = '아버지가 방에 들어가신다.'
twt = Twitter()
tagging = twt.pos(sentence)
tagging
{% endhighlight %}

```
>>> tagging
[('아버지', 'Noun'), ('가', 'Josa'), ('방', 'Noun'), ('에', 'Josa'), ('들어가신', 'Verb'), ('다', 'Eomi'), ('.', 'Punctuation')]
```

<figcaption class="caption">오!!!!!!!!</figcaption>

<br>

그러면 코멘드 창에서 코드를 치지 않고 파이썬 스크립트를 짜 놓고 .py 파일을 실행해도 같은 결과가 나오겠지?

![pic4](/assets/images/post/002/173_04.png)
<figcaption class="caption">또 왜...</figcaption>

<br>

또 konlpy가 패키지가 아니라는 오류다. 애초에 기대를 하면 안됐던 것 같다. 내가 쓴 코드를 한 줄씩 따라가보자면,
```
>>> quit()
```
"난 파이썬 커멘드창에서 나갈거야."

```
Hyeyeonui-MacBook-Pro:~ hyeyeonyang$ cd ~/webscraping
```
"'아버지가 방에 들어가신다' 코드는 webscraping 폴더 안에 담겨있어. 거기로 가자."

```
Hyeyeonui-MacBook-Pro:webscraping hyeyeonyang$ python3 konlpy2.py
```
"아버지 코드가 담긴 파이썬 파일 이름은 konlpy2.py야. 파이썬3아 실행시켜줄래?"


```
Traceback (most recent call last):
  File "konlpy2.py", line 1, in <module>
    from konlpy.tag import Twitter
  File "/Users/hyeyeonyang/Webscraping/konlpy.py", line 3, in <module>
    from konlpy.tag import Twitter
ModuleNotFoundError: No module named 'konlpy.tag'; 'konlpy' is not a package
```
...? 또 코엔엘파이를 불러오지 못한다. 뭔가 충돌이 있거나, 아까 성공한 게 우연이었거나, 아니면 설마 webscraping 폴더로 이동해서 그런가? 에이 설마.

---

## 포맷할까 진짜

마침 내 컴퓨터가 너무 느려진 것 같기도 하고, 그깟 파워포인트 만들 때 렉이 걸려서 화났던 기억도 나고, 금주의 무료앱이라면 쓸 데 없어보여도 다운받아놨던 게 컴퓨터를 더럽혔던건가 싶기기도 하고, 이참에 새 마음 새 출발 하고 싶기도 하고. 왜 난 한 번에 뭐가 제대로 된 적이 없을까.

또 에러뜨면 R이나 엑셀로 갈아탈 마음으로 다시 한 번 파이썬 쉘에서 실행해봤다.

![pic5](/assets/images/post/002/173_05.png)
![pic6](/assets/images/post/002/173_06.png)
<figcaption class="caption">오!류!</figcaption>

<br>

아까와의 차이는 이거다. 처음 시도 때는 'hyeyeonyang' 폴더에서 python3을 실행시켰고 두 번째 시도 때는 'hyeyeonyang/webscraping' 폴더에서 python3을 실행시켰다. 이제 'hyeyeonyang' 폴더에서 위 코드를 재실행했을 때 오류가 없으면 파이썬 실행 전 경로지정이 문제인거다.

![pic7](/assets/images/post/002/173_07.png)
<figcaption class="caption">말도 안 돼. 진짜 이게 원인이라고?</figcaption>

---

## 마음을 가다듬고

지금까지 흐름으로 보면, konlpy2.py 파일을 'hyeyeonyang/webscraping'이 아닌 'hyeyeonyang'으로 옮겨서 `python3 konlpy2.py`를 입력했을 때 오류 없이 코드가 실행되어야 한다. 해보자.

![pic8](/assets/images/post/002/173_08.png)
<figcaption class="caption">소름.. 실행된다..</figcaption>

<br>

이게 왜 이런건지는 모르겠다. 왜 특정 폴더에선 파이썬 패키지를 불러올 수 없는거지? 내가 이 폴더에 제한을 걸어두지도 않았고, 다른 패키지들은 다 이 폴더에서 문제없이 불러오고 있는데 도대체 왜?

---
